core/src/test/scala/unit/kafka/server/AlterReplicaDirRequestTest.scala

```diff
@@ -0,0 +1,82 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package kafka.server
+
+import kafka.network.SocketServer
+import kafka.utils._
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.protocol.{ApiKeys, Errors}
+import org.apache.kafka.common.requests._
+import org.junit.Assert._
+import org.junit.Test
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+class AlterReplicaDirRequestTest extends BaseRequestTest {
+
+  override def numBrokers: Int = 1
+  override def logDirCount: Int = 5
+
+  val topic = "topic"
+
+  @Test
+  def testAlterReplicaDirRequestBeforeTopicCreation() {
```

Comment:
I think it is OK to have both. If `AlterReplicaDirRequestTest.testAlterReplicaDirRequestBeforeTopicCreation` passes but `AdminClientIntegrationTest.testAlterReplicaLogDirBeforeTopicCreation` fails, it suggests that something is wrong in the KafkaAdminClient. This seems to follow the existing pattern -- tests such as `DeleteTopicsRequestTest` and `CreateTopicsRequestTest` probably would not be needed for the same reason given that we already have tests for higher level APIs that use them (if not then we should add the test for higher level APIs that uses them).

But I am not strong on this. I will remove this test if you think it is not necessary.